Missing requirement: org.eclipse.jface 3.39.0.v20251219-0945 requires 'osgi.ee; (&(osgi.ee=JavaSE)(version=21))' but it could not be found

[Eclipse Platform has updated BREE](https://github.com/eclipse-platform/eclipse.platform.ui/pull/3592)

[RCPTT build is failing](https://ci.eclipse.org/rcptt/view/active/job/master/2320/execution/node/40/log/) with:

```
[INFO] -------< org.eclipse.rcptt:org.eclipse.rcptt.tesla.nebula.impl >--------
 [INFO] Building org.eclipse.rcptt.tesla.nebula.impl 2.8.0-SNAPSHOT      [32/88]
 [INFO]   from /home/jenkins/agent/workspace/master/modules/nebula/plugins/org.eclipse.rcptt.tesla.nebula.impl/pom.xml
 [INFO] ---------------------------[ eclipse-plugin ]---------------------------
 [ERROR] org.eclipse.jface 3.39.0.v20251219-0945 requires Execution Environment that matches (&(osgi.ee=JavaSE)(version=21)) but the current resolution context uses [a.jre.javase 17.0.0]
 [INFO] {osgi.os=win32, osgi.ws=win32, org.eclipse.update.install.features=true, osgi.arch=x86_64, org.eclipse.update.install.sources=true}
 [ERROR] Cannot resolve project dependencies:
 [ERROR]   Software being installed: org.eclipse.rcptt.tesla.nebula.impl 2.8.0.qualifier
 [ERROR]   Missing requirement: org.eclipse.jface 3.39.0.v20251219-0945 requires 'osgi.ee; (&(osgi.ee=JavaSE)(version=21))' but it could not be found
 [ERROR]   Cannot satisfy dependency: org.eclipse.rcptt.tesla.jface.aspects 2.8.0.202512171837 depends on: osgi.bundle; org.eclipse.jface 0.0.0
 [ERROR]   Cannot satisfy dependency: org.eclipse.rcptt.tesla.nebula.impl 2.8.0.qualifier depends on: osgi.bundle; org.eclipse.rcptt.tesla.jface.aspects [2.8.0,3.0.0)
```

Problems:
- If I update Nebula and JFace support's BREE, the error will be fixed, but we will lose formal support of Java-17 in AUTs (which is unacceptable).
- Modern Nebula requires SWT 3.126, but the oldest supported runtime provides only SWT 3.118. Current build configuration makes it impossible to support both latest Nebula and older Eclipse Platform.

--- a/releng/ide/pom.xml
+++ b/releng/ide/pom.xml
@@ -34,6 +34,7 @@
         <groupId>${tycho-groupid}</groupId>
         <artifactId>target-platform-configuration</artifactId>
         <configuration>
+          <executionEnvironment>JavaSE-21</executionEnvironment>
           <environments>
             <environment>
               <os>win32</os>

--- a/releng/pom.xml
+++ b/releng/pom.xml
@@ -236,10 +236,6 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
-        <configuration>
-          <source>1.7</source>
-          <target>1.7</target>
-        </configuration>
       </plugin>
       <plugin>
         <groupId>${tycho-groupid}</groupId>

--- a/releng/runtime/pom.xml
+++ b/releng/runtime/pom.xml
@@ -85,6 +85,7 @@ documentation -->
 				<artifactId>target-platform-configuration</artifactId>
 				<configuration>
 					<requireEagerResolve>true</requireEagerResolve> <!--aspectj-maven-plugin fails without prepared Target Platform-->
+					<executionEnvironment>JavaSE-17</executionEnvironment>
 				</configuration>
 			</plugin>
 		</plugins>

